compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3282,15 +3282,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let val = if let Ok(base) = self.tcx.sess.source_map().span_to_snippet(base.span)
             && base.len() < 20
         {
-            format!("`{base}`")
+            format!("{base}")
         } else {
             "the value".to_string()
         };
         err.multipart_suggestion(
-            format!("{val} is a raw pointer; try dereferencing it"),
+            format!("`{val}` is a raw pointer; try dereferencing it"),
             vec![
-                (base.span.shrink_to_lo(), "(*".to_string()),
-                (base.span.shrink_to_hi(), ")".to_string()),
+                (base.span.shrink_to_lo(), "(*".into()),
+                (base.span.to(field.span.shrink_to_lo()), format!("{val}).")),
```

Comment:
You can't use `val` for suggestions, see above, if it's `>=20` bytes it gets replaced with `"the value"`.
```suggestion
                (base.span.between(field.span), format!(").")),
```